counsel.el

```diff
@@ -1322,34 +1322,38 @@ files in a project.")
        (setq cmd counsel-git-grep-cmd-default)))
     (cons proj cmd)))
 
-(defun counsel--call (&rest program-and-args)
-  (let ((stderr-file (make-temp-file "counsel-call-stderr")))
+(defun counsel--call (&rest command)
+  "Synchronously call COMMAND and return its output as a string.
+COMMAND comprises the program name followed by its arguments, as
+in `make-process'.  Signal `file-error' and emit a warning if
+COMMAND fails.  Obey file handlers based on `default-directory'."
+  (let ((stderr (make-temp-file "counsel-call-stderr-")))
     (unwind-protect
          (with-temp-buffer
-           (let ((res (apply #'call-process (car program-and-args)
-                             nil (list t stderr-file)
-                             nil (cdr program-and-args))))
-             (if (= 0 res)
-                 (buffer-substring (point-min)
-                                   (- (point-max)
-                                      (if (= ?\n (char-before (point-max)))
-                                          1
-                                        0)))
-               (error "%S exited with %d\n%s"
-                      (mapconcat #'identity program-and-args " ")
-                      res
-                      (with-temp-buffer
-                        (insert-file-contents stderr-file)
-                        (buffer-string))))))
-      (delete-file stderr-file))))
+           (let ((res (apply #'process-file (car command) nil
+                             (list t stderr) nil (cdr command))))
+             (unless (eq res 0)
+               (setq res (list 'file-error
+                               (mapconcat #'identity `(,@command "failed") " ")
+                               res))
+               (lwarn 'ivy :warning "%s"
+                      (apply #'concat
+                             (error-message-string res)
+                             (when (file-readable-p stderr)
+                               (insert-file-contents stderr nil nil nil t)
+                               (list "\n" (buffer-string)))))
+               (signal (car res) (cdr res)))
+             (buffer-substring (point-min) (- (point) (if (bolp) 1 0)))))
+      (when (file-exists-p stderr)
+        (delete-file stderr)))))
```

Comment:
Same here, just delete:

```
(ignore-errors (delete-file stderr))
```

Reply:
No need for `ignore-errors`; `delete-file` does not signal an error when given a non-existent file.

Reply:
See the latest commit.